fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializationContext.java

```diff
@@ -150,13 +155,17 @@ public void setSuccess(boolean success) {
     /**
      * recordFailReason
      */
-    public void recordFailReason(ObjectId objectId, Pair<String, String> summaryAndReason) {
+    public void recordFailReason(StructInfo structInfo, Pair<String, String> summaryAndReason) {
```

Comment:
maybe use this signature is better
```suggestion
    public void recordFailReason(StructInfo structInfo, String summary, Supplier<String> reasonSupplier) {
    ...
    String reason = enableRecordFailureDetail ? reasonSupplier.get() : ""
    ...
```

Reply:
have fixed it